src/auth/baseexternalclient.ts

```diff
@@ -708,4 +714,77 @@ export abstract class BaseExternalAccountClient extends AuthClient {
   protected getTokenUrl(): string {
     return this.tokenUrl;
   }
+
+  /**
+   * Returns the workforce identity pool-id if it is determinable
```

Comment:
nit: I think using "pool id" in the comments instead of "pool-id" would improve readability. It also makes searching the codebase easier, since we usually search for the term without a hyphen.